src/screens/discover/components/NetworkSwitcher.tsx

```diff
@@ -746,28 +755,24 @@ function Sheet({ children, header, onClose }: PropsWithChildren<{ header: ReactE
   );
 }
 
-export function NetworkSelector({
-  onClose,
-  multiple,
-}: {
-  onClose: (selected: ChainId[] | 'all') => void;
-  onSelect: VoidFunction;
-  multiple?: boolean;
-}) {
+export function NetworkSelector({ onClose }: { onClose: VoidFunction }) {
   const editing = useSharedValue(false);
-  const selected = useSharedValue<ChainId[] | 'all'>([]);
 
-  const close = () => {
-    'worklet';
-    runOnJS(onClose)(selected.value);
-  };
+  const network = useTrendingTokensStore(state => state.network);
+  const selected = useSharedValue<ChainId | undefined>(network);
+  const setNetwork = useTrendingTokensStore(state => state.setNetwork);
+
+  useSyncSharedValue<ChainId | undefined>({
+    sharedValue: selected,
+    syncDirection: 'sharedValueToState',
+    state: network,
+    setState: setNetwork,
+  });
```

Comment:
I'm not against doing this here, but just for context on the tradeoff, every selection change is gonna rerender/fetch the trending tokens screen behind the wallet switcher

so close and already very likely have the updated list data, but do more work while selecting 
vs 
only update the list when it's visible

Reply:
removed this also, ty for the tip